### PR TITLE
Bump kube-state-metrics to v2.0.0-rc.1

### PR DESCRIPTION
**Pull request by lilic**
* [x] I added CHANGELOG entry for this change.
* [ ] No user facing changes, so no entry in CHANGELOG was needed.

/hold

### CODE REVIEW

**Comment by lilic** [comment]
Bump PR -> https://github.com/openshift/kube-state-metrics/pull/47

File: assets/alertmanager/pod-disruption-budget.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+apiVersion: policy/v1beta1
```
Comment (lilic) [inline assets/alertmanager/pod-disruption-budget.yaml:1]: Is this something we want, this was generated by bumping kube-prom I believe.

File: assets/prometheus-k8s/pod-disruption-budget.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+apiVersion: policy/v1beta1
```
Comment (lilic) [inline assets/prometheus-k8s/pod-disruption-budget.yaml:1]: Same goes for this file.

File: assets/alertmanager/pod-disruption-budget.yaml (hunk 1 above)
Comment (simonpasquier) [reply assets/alertmanager/pod-disruption-budget.yaml:1]: I believe it is the fix for https://bugzilla.redhat.com/show_bug.cgi?id=1933847

**Comment by lilic** [comment]
Ready for review, but not for merge, hold still applies.

**Comment by lilic** [comment]
Waiting for https://github.com/openshift/kube-state-metrics/pull/47

**Comment by simonpasquier** [comment]
/lgtm

Though it seems backwards, we need this PR to be merged before the bump to kube-state-metrics v2 (see https://github.com/openshift/kube-state-metrics/pull/47#issuecomment-810188620). Because of metric changes, A few rules using kube-state-metrics wouldn't provide results once this PR get merged but the plan is to merge https://github.com/openshift/kube-state-metrics/pull/47 right after so the discrepancy shouldn't be noticeable.

**Comment by simonpasquier** [comment]
The upgrade job fails because the service account don't have permissions to list the ingress resources. I'll investigate later.

```
E0330 15:28:28.661842       1 reflector.go:178] k8s.io/kube-state-metrics/internal/store/builder.go:346: Failed to list *v1beta1.Ingress: ingresses.extensions is forbidden: User "system:serviceaccount:openshift-monitoring:kube-state-metrics" cannot list resource "ingresses" in API group "extensions" at the cluster scope
E0330 15:28:29.692772       1 reflector.go:178] k8s.io/kube-state-metrics/internal/store/builder.go:346: Failed to list *v1beta1.Ingress: ingresses.extensions is forbidden: User "system:serviceaccount:openshift-monitoring:kube-state-metrics" cannot list resource "ingresses" in API group "extensions" at the cluster scope
E0330 15:28:31.627119       1 reflector.go:178] k8s.io/kube-state-metrics/internal/store/builder.go:346: Failed to list *v1beta1.Ingress: ingresses.extensions is forbidden: User "system:serviceaccount:openshift-monitoring:kube-state-metrics" cannot list resource "ingresses" in API group "extensions" at the cluster scope
```

**Comment by simonpasquier** [comment]
/close
superseded by #1099

**Pull request closed without merging**